src/Swarm/Game/Location.hs

```diff
@@ -139,6 +141,28 @@ cardinalDirs =
 toDirection :: Heading -> Maybe Direction
 toDirection v = M.lookup v cardinalDirs
 
+-- | Example:
+--      DWest `relativeTo` DSouth == DRight
+relativeTo :: AbsoluteDir -> AbsoluteDir -> RelativeDir
+relativeTo targetDir referenceDir =
+  [DForward, DLeft, DBack, DRight] !! indexDiff
+ where
+  enumCount = length (Util.listEnums :: [AbsoluteDir])
+  indexDiff = (((-) `on` fromEnum) targetDir referenceDir + enumCount) `mod` enumCount
+
+-- | Logic adapted from:
+-- https://gamedev.stackexchange.com/questions/49290/#comment213403_49300
+nearestDirection :: Heading -> AbsoluteDir
+nearestDirection coord =
+  -- Using `cycle` obviates the `mod` operation
+  cycle orderedDirs !! index
```

Comment:
Personally I think just using `mod` on the rounded index would be clearer, especially since we don't have to worry about adding `enumCount` before performing the `mod`.